src/Language/PureScript/TypeChecker/Subsumption.hs

```diff
@@ -94,6 +94,8 @@ subsumes' mode (KindedType ty1 _) ty2 =
   subsumes' mode ty1 ty2
 subsumes' mode ty1 (KindedType ty2 _) =
   subsumes' mode ty1 ty2
+subsumes' mode (ProxyType ty1) (ProxyType ty2) =
```

Comment:
We shouldn't have proxies for polymorphic things, so I don't know why we need this. In any case, I don't think it's true that `Proxy` is necessarily covariant.